physicalplan/scan/matrix_selector.go

```diff
@@ -27,10 +27,15 @@ type matrixScanner struct {
 	samples        *storage.BufferedSeriesIterator
 }
 
+// Injectable refers to functionality that can be optionally injected into
+// different sub-trees of query AST.
+type Injectable struct {
+	Call function.FunctionCall
+}
+
 type matrixSelector struct {
-	funcExpr *parser.Call
-	call     FunctionCall
 	storage  *seriesSelector
+	in       *Injectable
```

Comment:
Hm I don't understand why this change is necessary. We should always be able to inject a function in a matrix selector. Otherwise the return type will not match.

Reply:
So earlier when we only handled `parser.MatrixSelector` via `parser.Call`, we could always inject the function directly. But now that it might be called a separate operator using queries like `http_requests_total[30s]`, it might not always need a call.

Also, I wanted to create a mechanism to inject some modifications while creating a new operator recursively, and this seemed like a way to achieve that. This can be used in the future when we might want to create different modified operators based on queries. Wdyt? 🙂

Reply:
For a query like `http_requests_total[30s]`, the return type is a really a matrix, not a vector. So I don't think that query will fit into the model that we have now, and I suspect we might need a completely different operator for that case.

Let's maybe try to tackle that problem separately, and use this PR to enable function calls with an arbitrary number of parameters.

Reply:
Yup, I'll add it as a TODO and remove.

Reply:
Have modified this to only have `call` for now. Added TODO